src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -299,13 +299,47 @@ public void SendFunctionLoadRequests(ManagedDependencyOptions managedDependencyO
                     _eventSubscriptions.Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionLoadResponse)
                         .Subscribe((msg) => LoadResponse(msg.Message.FunctionLoadResponse), HandleWorkerFunctionLoadError));
                 }
-                foreach (FunctionMetadata metadata in _functions.OrderBy(metadata => metadata.IsDisabled()))
+
+                // Check if the worker supports this feature
+                bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.AcceptsListOfFunctionLoadRequests));
+                if (capabilityEnabled)
+                {
+                    SendFunctionLoadRequestsList(_functions, managedDependencyOptions);
+                }
+                else
                 {
-                    SendFunctionLoadRequest(metadata, managedDependencyOptions);
+                    foreach (FunctionMetadata metadata in _functions.OrderBy(metadata => metadata.IsDisabled()))
+                    {
+                        SendFunctionLoadRequest(metadata, managedDependencyOptions);
+                    }
                 }
             }
         }
 
+        internal void SendFunctionLoadRequestsList(IEnumerable<FunctionMetadata> functions, ManagedDependencyOptions managedDependencyOptions)
+        {
+            _functionLoadRequestResponseEvent = _metricsLogger.LatencyEvent(MetricEventNames.FunctionLoadRequestResponse);
+
+            // send load requests for the registered functions
```

Comment:
Should we also do `_workerChannelLogger.LogDebug` similar to what happens in line 386? If so, your UT can be fixed to verify that as the expected log message.